File: ios/App/App/IonicNativePluginExample.m
//
//  IonicNativePluginExample.m
//  App
//
//  Created by James Haworth Wheatman on 05/06/2021.
//

#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(IonicNativePluginExample, "IonicNativePluginExample",
           CAP_PLUGIN_METHOD(NativeMethod, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(NotifyListeners, CAPPluginReturnPromise);
)

File: ios/App/App/IonicNativePluginExample.swift
//
//  IonicNativePluginExample.swift
//  App
//
//  Created by James Haworth Wheatman on 05/06/2021.
//

import Foundation
import Capacitor

@objc(IonicNativePluginExample)
public class IonicNativePluginExample: CAPPlugin {

    @objc func NativeMethod(_ call: CAPPluginCall) {
        call.resolve(["message": "Hello iOS user!"])
    }

    @objc func NotifyListeners(_ call: CAPPluginCall) {
        self.notifyListeners(
            "EVENT_LISTENER_NAME",
            data: ["message": "Hello iOS user!"]
        )
    }

}
